src/common/bin_serde.rs

```diff
@@ -0,0 +1,99 @@
+//! Binary serialization, and an implementation over Unix pipes.
+use std::{
+    io::{self, Read, Write},
+    marker::PhantomData,
+    os::unix::net::UnixStream,
+};
+
+/// Serialization/deserialization trait using a byte array as storage.
+pub trait BinSerDe<const N: usize> {
+    fn serialize(&self) -> [u8; N];
+    fn deserialize(val: [u8; N]) -> Self;
+}
+
+/// A pipe abstracting over a [UnixStream] with easier
+/// binary serialization, to help with the buffer sizes and ser/de steps.
+/// Uses [UnixStream::pair()].
+pub fn pair<const N1: usize, T1: BinSerDe<N1>, const N2: usize, T2: BinSerDe<N2>>(
```

Comment:
I think this function should be associated to `BinPipe` so you can call it as `BinPipe::pair` and you don't need to import  both `BinPipe` and `pair` later.